api/authenticator.py

```diff
@@ -1362,12 +1369,17 @@ def remote_patron_lookup(self, patron_or_patrondata):
             return None
         if (isinstance(patron_or_patrondata, PatronData)
             or isinstance(patron_or_patrondata, Patron)):
+
+
+
             return patron_or_patrondata
         raise ValueError(
             "Unexpected object %r passed into remote_patron_lookup." %
             patron_or_patrondata
         )
 
+    _remote_patron_lookup = remote_patron_lookup
```

Comment:
Without going through this code more carefully, this is confusing. I don't understand the purposes of these two methods - sometimes they are the same, sometimes one calls the other. Why do we need two?